File: java/test/Ice/location/ServerLocatorRegistry.java
// **********************************************************************
//
// Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

public class ServerLocatorRegistry extends Test._TestLocatorRegistryDisp
{
    public void
    setAdapterDirectProxy_async(Ice.AMD_LocatorRegistry_setAdapterDirectProxy cb, String adapter,
                                Ice.ObjectPrx object, Ice.Current current)
    {
        _adapters.put(adapter, object);
        cb.ice_response();
    }

    public void
    setReplicatedAdapterDirectProxy_async(Ice.AMD_LocatorRegistry_setReplicatedAdapterDirectProxy cb, String adapter, 
                                          String replica, Ice.ObjectPrx object, Ice.Current current)
    {
        _adapters.put(adapter, object);
        _adapters.put(replica, object);
        cb.ice_response();
    }

    public void
    setServerProcessProxy_async(Ice.AMD_LocatorRegistry_setServerProcessProxy cb, String id, Ice.ProcessPrx proxy,
                                Ice.Current current)
    {
    }

    public void
    addObject(Ice.ObjectPrx object, Ice.Current current)
    {
        _objects.put(object.ice_getIdentity(), object);
    }

    public Ice.ObjectPrx
    getAdapter(String adapter)
        throws Ice.AdapterNotFoundException
    {
        Ice.ObjectPrx obj = (Ice.ObjectPrx)_adapters.get(adapter);
        if(obj == null)
        {
            throw new Ice.AdapterNotFoundException();
        }
        return obj;
    }

    public Ice.ObjectPrx
    getObject(Ice.Identity id)
        throws Ice.ObjectNotFoundException
    {
        Ice.ObjectPrx obj = (Ice.ObjectPrx)_objects.get(id);
        if(obj == null)
        {
            throw new Ice.ObjectNotFoundException();   
        }
        return obj;
    }

    private java.util.HashMap _adapters = new java.util.HashMap();
    private java.util.HashMap _objects = new java.util.HashMap();
}

File: cs/test/Ice/location/ServerLocatorRegistry.cs
// **********************************************************************
//
// Copyright (c) 2003
// ZeroC, Inc.
// Billerica, MA, USA
//
// All Rights Reserved.
//
// Ice is free software; you can redistribute it and/or modify it under
// the terms of the GNU General Public License version 2 as published by
// the Free Software Foundation.
//
// **********************************************************************

using System.Collections;

public class ServerLocatorRegistry : Ice.LocatorRegistry_Disp
{
    public ServerLocatorRegistry()
    {
        _adapters = new Hashtable();
	_objects = new Hashtable();
    }

    public override void setAdapterDirectProxy(string adapter, Ice.ObjectPrx obj, Ice.Current current)
    {
        _adapters[adapter] = obj;
    }

    public override void setServerProcessProxy(string id, Ice.ProcessPrx proxy, Ice.Current current)
    {
	// TODO: missing implementation
    }

    public virtual Ice.ObjectPrx getAdapter(string adapter)
    {
        object obj = _adapters[adapter];
        if(obj == null)
        {
            throw new Ice.AdapterNotFoundException();
        }
        return (Ice.ObjectPrx)obj;
    }
    
    public virtual Ice.ObjectPrx getObject(Ice.Identity id)
    {
        object obj = _objects[id];
        if(obj == null)
        {
            throw new Ice.ObjectNotFoundException();
        }
        return (Ice.ObjectPrx)obj;
    }
    
    public virtual void addObject(Ice.ObjectPrx obj)
    {
        _objects[obj.ice_getIdentity()] = obj;
    }
    
    private Hashtable _adapters;
    private Hashtable _objects;
}
